cw/capture.py

```diff
@@ -1616,14 +1616,43 @@ def capture_otbn_vertical_batch(ot, ktp, capture_cfg, scope_type, device_cfg):
         ot.target.simpleserial_write("s", capture_cfg["batch_prng_seed"].to_bytes(4, "little"))
         time.sleep(0.3)
 
-        # set ecc256 fixed seed
-        seed_fixed = ktp.next_key()
+        # Generate fixed constants for all traces of the keygen operation.
+
+        if capture_cfg["test_type"] == 'KEY':
+            # In fixed-vs-random KEY mode we use two fixed constants:
+            #    1. C - a 320 bit constant redundancy
+            #    2. fixed_number - a 256 bit number used to derive the fixed key
+            #                      for the fixed set of measurements. Note that in
+            #                      this set, the fixed key is equal to
+            #                      (C + fixed_number) mod curve_order_n
+            C = ktp.next_key()
+            if len(C) != seed_bytes:
+                raise ValueError(f'Fixed seed length is {len(C)}, expected {seed_bytes}')
+            ktp.key_len = key_bytes
+            fixed_number = ktp.next_key()
+            if len(fixed_number) != key_bytes:
+                raise ValueError(f'Fixed key length is {len(fixed_number)}, expected {key_bytes}')
+            ktp.key_len = seed_bytes
+            C_int = int.from_bytes(C, byteorder='little')
+            seed_fixed_int = C_int + int.from_bytes(fixed_number, byteorder='little')
```

Comment:
I didn't think it was important to check for this because we are adding a 320-bit and 256-bit number. The only way an overflow can happen is if the first 64 MSBs of C_int are all 1s, which can only happen with a very small probability.